src/Sql/dbo/Stored Procedures/User_ReadBySsoUserOrganizationIdExternalId.sql

```diff
@@ -0,0 +1,17 @@
+﻿CREATE PROCEDURE [dbo].[User_ReadBySsoUserOrganizationIdExternalId]
+    @OrganizationId UNIQUEIDENTIFIER,
+    @ExternalId NVARCHAR(50)
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        U.*
+    FROM
+        [dbo].[UserView] U
+    INNER JOIN
+        [dbo].[SsoUser] SU ON SU.[UserId] = U.[Id]
+    WHERE
+        SU.[OrganizationId] = @OrganizationId
```

Comment:
Since `OrganizationId` can be null, and I'm assuming we would want an exact match on Org and External Id, I would also imagine we would want External Id to be unique for non-organizational user mappings as well... To that end, if OrganizationId is `null` in the DB, and `@OrganizationId` is `null` being passed in, we still won't find that entry because in SQL Server, `null != null`, you have to use a `COALESCE()` or `ISNULL()` function for nullable comparison when `null` is a valid combination.

Reply:
Fixed